rfcs/0000-image-build-node-selection.md

```diff
@@ -0,0 +1,85 @@
+
+**Outcome:**
+
+kpack Image and Build resources should expose a mechanism in the v1alpha2 api to provide nodeSelectors, nodeAffinity, and tolerations. These resources should be unmodified from the core v1 Pod api. The contents of these fields should be passed to the underlying build pod without modification except when adding the `kubernetes.io/os` nodeSelector. 
+
+On the Image resource these fields should exist under `.spec.build` on the Image spec.
+
+```yaml
+apiVersion: kpack.io/v1alpha2
+kind: Image
+metadata:
+  name: sample
+spec:
+  tag: sample/image
+  ...
+  build:
+    tolerations:
+    - key: "key1"
+      operator: "Exists"
+      effect: "NoSchedule"
+    nodeSelector:
+      disktype: ssd
+    affinity:
+      nodeAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+          - matchExpressions:
+            - key: kubernetes.io/e2e-az-name
+              operator: In
+              values:
+              - e2e-az1
+              - e2e-az2
+```
+
+On the Build resource these fields should exist directly on `.spec` as top level fields on the Build spec.
+
+```yaml
+apiVersion: kpack.io/v1alpha2
+kind: Build
+metadata:
+  name: sample
+spec:
+  tags:
+  - sample/image
+  ...
+  tolerations:
+  - key: "key1"
+    operator: "Exists"
+    effect: "NoSchedule"
+  nodeSelector:
+    disktype: ssd
+  affinity:
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+        - matchExpressions:
+          - key: kubernetes.io/e2e-az-name
+            operator: In
+            values:
+            - e2e-az1
+            - e2e-az2
+```
+
+With the Image and Build resources allowing users to set node tolerrations, kpack should not attempt to calculate windows node tolerations indepdendently and instead should require windows users to explicitly set the necessary tolerations. 
+
+**Complexity:**
+
+Implementation of this functionality should be of minimal complexity because the contents of the additional fields can be passed directly to the build pod without modification. This functionality will also allow kpack to remove unnecessary complexity in kpack windows toleration selection logic. 
+
+**Prior Art:**
+
+* knative?
+
+**Alternatives:**
+
+* kpack could provide [node selection fields on the Builder and ClusterBuilder resources](https://github.com/pivotal/kpack/issues/621#issuecomment-892593799) and allow these fields to be applied to all corresponding Image resources. This is something we could explore in a future RFC. Currently, non builder fields are unprecedented on the Builder and ClusterBuilder resources. 
+
+**Risks:**
+
+This will effectively require Windows Image and Build resources to be created in the v1alpha2 api because the tolerations are not accessible in the v1alpha1 api. This is a minor risk because kpack has very few current windows build users. 
```

Comment:
As a part of this rfc we wouldn't be removing the windows toleration logic in v1alpha1 though right? Wouldn't they still be able to be created using the old logic? Why does this restrict windows image/builds to v1alpha2?